lib/Doctrine/ORM/EntityManager.php

```diff
@@ -145,6 +146,13 @@
     /** @var Cache The second level cache regions API. */
     private $cache;
 
+    /**
+     * This flag shows that flush is in progress
+     *
+     * @var bool
+     */
+    private $flushInProgress = false;
```

Comment:
As discussed in https://github.com/doctrine/doctrine2/pull/7448#pullrequestreview-173711503, this should be moved into the `UnitOfWork`.

Reply:
Please also adapt the naming, we don't "flush" in the UoW but rather "commit"